### PR TITLE
Chat features

### LINKED ISSUE #31: add chat features

### PR DESCRIPTION
**Pull request by MohammedMahani**

### CODE REVIEW

File: server/controllers/errors/serverError.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const { CustomeError } = require('../../utils');
 
 // eslint-disable-next-line no-unused-vars
 const serverError = (err, req, res, next) => {
+  console.log(err);
```
Comment (abdullah-alsharif) [inline server/controllers/errors/serverError.js]: remove console.log here

File: client/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "react-dom": "^18.2.0",
     "react-oauth": "^0.0.1",
     "react-router-dom": "^6.13.0",
+    "url": "^0.11.1",
```
Comment (devstm) [inline client/package.json]: uneeded backage

File: client/src/components/ChatBox/style.modules.css [hunk 3]
```diff
@@ -150,14 +150,28 @@ main.ant-layout-content.css-dev-only-do-not-override-12jzuas {
 
 .controler {
     display: flex;
-    justify-content: space-between;
+    /* justify-content: space-between; */
```
Comment (devstm) [inline client/src/components/ChatBox/style.modules.css]: dilete line or uncoment

File: client/src/components/Message/index.tsx [hunk 4]
```diff
@@ -1,19 +1,221 @@
...
-            </div>
-        </div>
-    );
+  const [translatedText, setTranslatedText] = useState('');
+  const [translateFrom, setTranslateFrom] = useState('en-GB');
+  const [translateTo, setTranslateTo] = useState('ar-SA');
+  const [copy, setCopy] = useState(false);
+  const [translate, setTranslate] = useState(false);
+
+  const countries = {
```
Comment (devstm) [inline client/src/components/Message/index.tsx]: put this opject in new file

File: server/controllers/errors/serverError.js (hunk 1 above)
Comment (devstm) [inline server/controllers/errors/serverError.js]: remove this console

File: server/config/app.config.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 require('dotenv').config();
 
 const app = {
-  port: 3000,
+  port: 5000,
```
Comment (devstm) [inline server/config/app.config.js]: undo this change

File: package.json [hunk 6]
```diff
@@ -31,6 +31,7 @@
     "dotenv": "^16.0.3",
     "express": "^4.18.2",
     "google-auth-library": "^8.8.0",
+    "google-tts-api": "^2.0.2",
```
Comment (devstm) [inline package.json]: is it needed

**Review by devstm — CHANGES_REQUESTED** [review]

File: client/src/components/Message/index.tsx [hunk 7]
```diff
@@ -1,19 +1,123 @@
...
+  const handleTranslate = async () => {
+    const apiUrl = `https://api.mymemory.translated.net/get?q=${encodeURIComponent(
+      content
+    )}&langpair=${translateFrom}|${translateTo}`;
+  
+    try {
+      const res = await axios.get(apiUrl);
+      const translation = res.data.responseData.translatedText;
+      setTranslatedText(translation);
+      if(translate){
```
Comment (AyaQunoo) [inline client/src/components/Message/index.tsx]: no need for if statment you can setTranslate(!translate);

File: client/src/components/Message/index.tsx [hunk 8]
```diff
@@ -1,19 +1,123 @@
...
-import { TranslationOutlined, SoundOutlined } from '@ant-design/icons';
+import React, { useState } from 'react';
+import axios from 'axios';
+import {
+  TranslationOutlined,
+  SoundOutlined,
+  CopyOutlined,
+} from '@ant-design/icons';
 import { MessageObjectType } from '../../utils';
+import { Select,Collapse, Alert } from 'antd';
```
Comment (devstm) [inline client/src/components/Message/index.tsx:9]: antd should be in top of utils

**Review by devstm — APPROVED** [review]

**Review by abdullah-alsharif — APPROVED** [review]

**Review by devstm — APPROVED** [review]

**Pull request merged**